jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -654,11 +659,22 @@ private String setupSettingFile(@Nonnull Collection<Credentials> credentials) th
             return settingsDest.getRemote();
         }
 
-        // Settings provided by the global maven configuration
-        SettingsProvider settingsProvider = GlobalMavenConfig.get().getSettingsProvider();
+        SettingsProvider settingsProvider;
+
+        MavenConfigFolderOverrideProperty overrideProperty = getMavenConfigOverrideProperty();
+        if (overrideProperty != null) {
+            // Settings overriden by a folder property
+            console.format("[withMaven] using overriden Maven settings by folder '%s'%n", overrideProperty.getOwner().getDisplayName());
+            settingsProvider = overrideProperty.getSettings();
+        } else {
+            console.format("[withMaven] using Maven settings provided by the Jenkins global configuration%n");
+            // Settings provided by the global maven configuration
+            settingsProvider = GlobalMavenConfig.get().getSettingsProvider();
+        }
+
         if (settingsProvider instanceof MvnSettingsProvider) {
             MvnSettingsProvider mvnSettingsProvider = (MvnSettingsProvider) settingsProvider;
-            console.format("[withMaven] use Maven settings provided by the Jenkins global configuration '%s' %n", mvnSettingsProvider.getSettingsConfigId());
+            console.format("[withMaven] use Config File Provide maven settings file '%s' %n", mvnSettingsProvider.getSettingsConfigId());

```

Comment:
Shall we indicate if the config has specified "globally" or by a "folder"?

Reply:
It is in the lines above, where your previous comment, do you want it here too?